tensorflow_datasets/scripts/documentation/generate_dataframe.py

```diff
@@ -0,0 +1,213 @@
+# limitations under the License.
+
+r"""Script which generates datasets dataframes.
+"""
+
+import functools
+import itertools
+import multiprocessing
+import os
+import imageio
+import tempfile
+import traceback
+from typing import List, Optional
+
+from absl import app
+from absl import flags
+from absl import logging
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow_datasets.core import lazy_imports_lib
+from tensorflow_datasets.core import features as features_lib
+from tensorflow_datasets.core.visualization import visualizer
+
+
+WORKER_COUNT_DATASETS = 10
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    'datasets', None,
+    'Comma separated list of datasets to generates. None for all datasets.')
+flags.DEFINE_string(
+    'dst_dir', tfds.core.gcs_path('visualization/dataframe'),
+    'Destination dir to save the dataframe html.')
+flags.DEFINE_boolean(
+    'overwrite', False, 'If True, overwrite the existing visualizations.')
+
+
+# pylint: disable=logging-format-interpolation,logging-not-lazy
+
+
+def _log_exception(fn):
+  """Logs the exceptions from a `ThreadPoolExecutor`."""
+
+  @functools.wraps(fn)
+  def decorated(*args, **kwargs):
+    try:
+      return fn(*args, **kwargs)
+    except Exception:  # pylint: disable=broad-except
+      err_str = traceback.format_exc()
+      logging.error(f'Exception occurred for {args}, {kwargs}:\n' + err_str)
+      raise
+
+  return decorated
+
+
+@_log_exception
+def _generate_single_dataframe(full_name: str, dst_dir: str) -> None:
+  """Save the generated dataframe HTML for the dataset in dst_dir.
+
+  Args:
+    full_name: Name of the dataset to build `dataset`, `dataset/config`.
+    dst_dir: Destination where the dataset will be saved (as
+      `dataset-config-version`)
+  """
+  dst = full_name.replace('/', '-')
+  dst_filename = dst + '.html'
+  dst_path = os.path.join(dst_dir, dst_filename)
+  # If the html already exists, skip the html generation
+  if not FLAGS.overwrite and tf.io.gfile.exists(dst_path):
+    logging.info(f'Skipping dataframe for {full_name} (already exists)')
+    return
+
+  logging.info(f'Generating dataframe for {full_name}...')
+  # Load the dataset.
+  builder_name, _, version = full_name.rpartition('/')
+  builder = tfds.builder(f'{builder_name}:{version}')
+  split_names = list(builder.info.splits.keys())
+  if not split_names:
+    logging.info(f'Dataset `{full_name}` not generated.')
+    return
+  elif 'train' in split_names:
+    split = 'train'
+  else:
+    split = split_names[0]
+  ds = builder.as_dataset(split=split, shuffle_files=False)
+
+  if not tf.io.gfile.exists(dst_dir):
+    tf.io.gfile.makedirs(dst_dir)
+  try:
+    df = tfds.as_dataframe(ds.take(10), builder.info)
+  except Exception:  # pylint: disable=broad-except
+    logging.info(f'Dataframe not supported for dataset `{full_name}`')
+    return
+
+  # Add audio, image, and video to html
+  audio_keys = visualizer.extract_keys(
```

Comment:
I don't understand why you need this complicated code. Just use `df._repr_html_()` to get the HTML representation, like in https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/as_dataframe_test.py

Reply:
I thought we wanted to display the images in the dataframe and have embedded players for the audio and video. This essentially saves the image, audio, and video and inserts HTML tags for embedded players in the dataframe.

Reply:
I already implemented the HTML tag in https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/features/image_feature.py#L221
So just saving the HTML representation will work.

I would recommend you to try `tfds.as_numpy` on Colab with TFDS nightly and some images dataset, you'll see what I meant.

Reply:
For video, I was thinking about something like https://github.com/tensorflow/datasets/pull/2235

Reply:
Sounds good.